scipy/signal/tests/test_upfirdn.py

```diff
@@ -35,20 +35,30 @@
 import numpy as np
 from itertools import product
 
-from scipy._lib._array_api import xp_assert_close
+from scipy._lib._array_api import (
+    xp_assert_close, array_namespace, is_torch, is_array_api_strict
+)
 from pytest import raises as assert_raises
 import pytest
```

Comment:
minor nit: the `pytest` imports belong above scipy-internal ones, in case you make another commit you may want to tidy up a bit.